bindings/muen/muen-net.c

```diff
@@ -103,62 +145,80 @@ void generate_mac_addr(uint8_t *addr)
     addr[0] |= 0x02;
 }
 
-void net_init(void)
+static bool muen_net_dev_init(const char *name, struct muen_net_device *device,
+        struct solo5_net_info *info)
 {
-    char mac_str[18];
     const uint64_t epoch = muen_get_sched_start();
```

Comment:
I guess we could check `device->acquired` here as well?

Reply:
We could but it felt a bit redundant since the function is only called internally.

Reply:
Actually, disregard that last comment. At this point `acquired` has not been set yet. That only happens after everything succeeds at the end of `solo5_net_acquire`.

Reply:
So does that mean that we *should* check `acquired` here?

Reply:
Sorry for the confusing comments: no, I don't think it is necessary since the function is only used internally and `acquired` has not yet been set at this point. We could check for `!acquired` but not sure its worth it?